lib/bap/bap_self.ml

```diff
@@ -60,4 +60,66 @@ module Create() = struct
   let warning f = message Warning ~section:name f
   let error f = message Error ~section:name f
 
+  module Config = struct
+    let plugin_name = name
+    include Bap_config
+
+    let (/) = Filename.concat
+    let confdir = confdir / plugin_name
+    let conf_filename = confdir / "config"
+
+    let get_env_options () =
+      let prefix = "BAP_" ^ String.uppercase plugin_name ^ "_" in
+      let prefix_chop_key (k, v) =
+        match String.chop_prefix ~prefix k with
+        | Some k -> Some (k, v)
+        | None -> None in
+      let plugin_filter_map str =
+        match String.split str ~on:'=' with
+        | k :: vs -> prefix_chop_key (k, String.concat ~sep:"=" vs)
+        | [] -> None in
+      Unix.environment () |>
+      Array.to_list |>
+      List.filter_map ~f:plugin_filter_map
+
+    let conf_file_options () =
+      let string_splitter str =
+        match String.split str ~on:'=' with
+        | k :: vs -> k, String.concat ~sep:"=" vs
+        | [] -> invalid_arg "empty string" in
+      let split_filter = List.map ~f:string_splitter in
+      try
+        In_channel.with_file
+          conf_filename ~f:(fun ch -> In_channel.input_lines ch
+                                      |> split_filter)
+      with Sys_error _ -> []
+
+    let options () =
```

Comment:
This is _not_ just a concatenation. `(fun o (k, v) -> List.Assoc.add o k v)` does an over-writing add. Basically, it ensures that the environment configuration overriding happens. Concatenation would not do this (and would create duplicates, and I could not find any documentation on which order `List.Assoc` would `find` things when there are duplicates)